msal/src/main/java/com/microsoft/identity/client/AuthenticationResult.java

```diff
@@ -32,29 +32,52 @@
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 
+import static com.microsoft.identity.common.internal.authscheme.PopAuthenticationSchemeInternal.SCHEME_POP;
+
 /**
  * MSAL successful authentication result. When auth succeeds, token will be wrapped into the
  * {@link AuthenticationResult} and passed back through the {@link AuthenticationCallback}.
  */
 public final class AuthenticationResult implements IAuthenticationResult {
 
+    private static final String SCHEME_DELIMITER = " ";
+
     private final String mTenantId;
     private final AccessTokenRecord mAccessToken;
     private final IAccount mAccount;
+    private final String mAuthorizationHeader;
 
-    public AuthenticationResult(@NonNull final List<ICacheRecord> cacheRecords) {
+    AuthenticationResult(@NonNull final List<ICacheRecord> cacheRecords,
+                         @NonNull final String authHeader) {
         final ICacheRecord mostRecentlyAuthorized = cacheRecords.get(0);
         mAccessToken = mostRecentlyAuthorized.getAccessToken();
         mTenantId = mostRecentlyAuthorized.getAccount().getRealm();
         mAccount = AccountAdapter.adapt(cacheRecords).get(0);
+        mAuthorizationHeader = authHeader;
     }
 
     @Override
     @NonNull
     public String getAccessToken() {
+        if (mAuthorizationHeader.startsWith(SCHEME_POP + SCHEME_DELIMITER)) {
```

Comment:
I'm thinking if we should mint the AT in AuthenticationResultAdapter.adapt(), or even here. (instead of minting it as part of mAuthorizationHeader and pass it all the way up from the lower layer).

i.e. if we want to do here, we could probably do something like
if (mAccessToken.getAccessTokenType() == SCHEME_POP){
    return (IPopAuthenticationSchemeInternal) scheme.mintAccessToken(mAccessToken.getSecret());
}

return mAccessToken.getSecret();

If we do this in adapt(), then there is no need to pass the ITokenAuthenticationSchemeInternal object here.
also this means there is no need to add mAccessToken as part of the scheme object.